### PR TITLE
Puppet Scripts for OB 2.0.0

### PR DESCRIPTION
**Pull request by Dinushi**
## Puppet Scripts for OB 2.0.0

> The puppet scripts for Open Banking 2.0.0-M2 release deployment

**Issue link:** *required*

**Doc Issue:** *Optional, link issue from [documentation repository](https://github.com/wso2/docs-ob/issues)*

**Applicable Labels:** *Spec, product, version, type (specify requested labels)*

------

### Development Checklist

1. [ ] Built complete solution with pull request in place.
2. [ ] Ran checkstyle plugin with pull request in place.
3. [ ] Ran Findbugs plugin with pull request in place.
4. [ ] Ran FindSecurityBugs plugin and verified report.
5. [ ] Formatted code according to WSO2 code style.
6. [ ] Have you verify the PR does't commit any keys, passwords, tokens, usernames, or other secrets?
7. [ ] Migration scripts written (if applicable).
8. [ ] Have you followed secure coding standards in [WSO2 Secure Engineering Guidelines](http://wso2.com/technical-reports/wso2-secure-engineering-guidelines)? 

### Testing Checklist

1. [ ] Written unit tests.
2. [ ] Documented test scenarios(link available in guides).
3. [ ] Written automation tests (link available in guides).
4. [ ] Verified tests in multiple database environments (if applicable).
5. [ ] Verified tests in multiple deployed specifications (if applicable).
6. [ ] Tested with OBBI enabled  (if applicable).
7. [ ] Tested with specification regulatory conformance suites  (if applicable).

**Automation Test Details**

| Test Suite        | Test Script IDs   |
| ----------------- | ----------------- |
| Integration Suite | *TCXXXXX, TCXXXX* |

**Conformance Tests Details**

| Test Suite Name  | Test Suite Version | Scenarios  | Result   |
| ---------------- | ------------------ | ---------- | -------- |
| *Security Suite* | *VX.X*             | *Foo, Bar* | *Passed* |

## Resources

**Knowledge Base:** https://sites.google.com/wso2.com/open-banking/

**Guides:** https://sites.google.com/wso2.com/open-banking/developer-guides

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,44 +6,45 @@ This repository contains the Puppet modules for WSO2 Open Banking.
 
 1. Download the following zip files:<br>
 
-    * wso2-obkm-1.5.0.zip <br>
-    * wso2-obam-1.5.0.zip <br>
+    * wso2-obiam-2.0.0-M2.zip <br>
```
[the remark this answers is not in the document]
Comment (Dinushi) [reply README.md:9]: updated

**Review by amalkasubasinghe — APPROVED** [review]

File: modules/ob_common/manifests/params.pp [hunk 2]
```diff
@@ -45,64 +45,72 @@
...
-  $db_apimgt_stat_username          = 'wso2carbon'
-  $db_apimgt_stat_password          = 'wso2carbon'
-  $db_apimgt_stat_validation_query  = 'SELECT 1'
-  $db_apimgt_stat_driver            = 'org.h2.Driver'
+  $db_apimgt_url                     = 'jdbc:h2:repository/database/WSO2AM_DB;DB_CLOSE_ON_EXIT=FALSE'
+  $db_apimgt_username                = 'wso2carbon'
+  $db_apimgt_password                = 'wso2carbon'
+  $db_apimgt_validation_query        = 'SELECT 1'
+  # Use below for oracle
+  # $db_apimgt_validation_query      = 'SELECT 1 FROM DUAL'
```
Comment (VimukthiPerera) [inline modules/ob_common/manifests/params.pp:65]: Shall we add this to the Readme under a new topic - 'customizations'? The validation query is not the only thing that changes when you change the DB to Oracle.

File: modules/obiam/templates/carbon-home/repository/deployment/server/jaggeryapps/consentmgt/configs/conf.json.erb [hunk 3]
```diff
@@ -22,5 +22,5 @@
 		"actualLimit" : 10,
 		"offset": 0
 	},
-	"DeployedSpecification" : <%= @spec %>
-}
+	"DeployedSpecification" : "<%= @spec %>"
+}
```
Comment (VimukthiPerera) [inline modules/obiam/templates/carbon-home/repository/deployment/server/jaggeryapps/consentmgt/configs/conf.json.erb:26]: Add a newline at the EOF.

**Review by VimukthiPerera — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -6,44 +6,45 @@ This repository contains the Puppet modules for WSO2 Open Banking.
...
-    .com/technetwork/java/javase/downloads/java-archive-javase8-2177648.html) and copy the archive into the 
-    `<puppet_environment>/modules/ob_common/files/jdk` directory.<br> (please make sure that the name of the archive 
+    a. Download Java SE Development Kit 8u161 for Linux x64 from [here](https://www.oracle.com/technetwork/java/javase/downloads/java-archive-javase8-2177648.html) and copy the archive into the
+    `<puppet_environment>/modules/ob_common/files/jdk` directory of **Puppetmaster**. <br> (please make sure that the name of the archive 
     matches with the name of the extracted folder Ex: jdk1.8.0_161) <br>
     b. Reassign the *$jdk_name* variable in `<puppet_environment>/modules/ob_common/manifests/params.pp` to the name of the downloaded JDK distribution.
 
-3. Set up the databases by following the [Configuring Databases](https://docs.wso2
-.com/display/OB140/Configuring+Databases) documentation.
+3. Change the hostnames in `<puppet_environment>modules/ob_common/manifests/params.pp` of **Puppetmaster**.
```
Comment (DivyaPremanantha) [inline README.md:23]: Shall we have a backslash after <puppet_environment>

File: README.md [hunk 5]
```diff
@@ -6,44 +6,45 @@ This repository contains the Puppet modules for WSO2 Open Banking.
...
 
-4. Run the following profiles on the **Puppet agent**, by executing the following commands.
+4. Set up the databases in the **Puppet agent** by following the [Configuring Databases](https://docs.wso2.com/display/OB140/Configuring+Databases) documentation. <br>
+   Update the `<puppet_environment>modules/ob_common/manifests/params.pp` of **Puppetmaster** to point the created databases.
```
Comment (DivyaPremanantha) [inline README.md:26]: Same as https://github.com/wso2/puppet-open-banking/pull/25/files#r481695603

File: modules/obam/templates/carbon-home/bin/wso2server.sh.erb [hunk 6]
```diff
@@ -158,16 +158,27 @@ do
...
-          do
-            case "$profile" in
-              *Dprofile=*)
-                cd $(dirname "$0")
-                sh profileSetup.sh $profile
-                echo "Starting the server..."
-                ;;
-            esac
-          done
+      for option in $*; do
```
Comment (DivyaPremanantha) [inline modules/obam/templates/carbon-home/bin/wso2server.sh.erb:161]: Shall we format this loop

File: modules/obam/templates/carbon-home/bin/wso2server.sh.erb [hunk 7]
```diff
@@ -158,16 +158,27 @@ do
...
+        if [ "$option" = "--skipConfigOptimization" ] || [ "$option" = "-skipConfigOptimization" ] ||
+        [ "$option" = "skipConfigOptimization" ]; then
+          passedSkipConfigOptimizationOption=true
+          echo "Passed skipConfigOptimization Option: $passedSkipConfigOptimizationOption"
+        fi
+      done
+
+      for profile in $*; do
```
Comment (DivyaPremanantha) [inline modules/obam/templates/carbon-home/bin/wso2server.sh.erb:169]: Same as https://github.com/wso2/puppet-open-banking/pull/25/files#r481707058

File: modules/obiam/templates/carbon-home/repository/deployment/server/jaggeryapps/consentmgt/configs/conf.json.erb (hunk 3 above)
Comment (Dinushi) [reply modules/obiam/templates/carbon-home/repository/deployment/server/jaggeryapps/consentmgt/configs/conf.json.erb:26]: done!

File: modules/ob_common/manifests/params.pp (hunk 2 above)
Comment (Dinushi) [reply modules/ob_common/manifests/params.pp:65]: done! Mentioned the database type specific config changes in a separate section of the readMe file

File: README.md (hunk 4 above)
Comment (Dinushi) [reply README.md:23]: done!

File: README.md (hunk 5 above)
Comment (Dinushi) [reply README.md:26]: done!

File: modules/obam/templates/carbon-home/bin/wso2server.sh.erb (hunk 6 above)
Comment (Dinushi) [reply modules/obam/templates/carbon-home/bin/wso2server.sh.erb:161]: done!

File: modules/obam/templates/carbon-home/bin/wso2server.sh.erb (hunk 7 above)
Comment (Dinushi) [reply modules/obam/templates/carbon-home/bin/wso2server.sh.erb:169]: done!

**Comment by Dinushi** [comment]
Since this puppet-open-banking repository was recently been moved to this public location, I had to send the changes in this PR in a newly created PR.
I have resolved all the comments above in that PR (https://github.com/wso2/puppet-open-banking/pull/27).

Hence, I will close this PR.

**Pull request closed without merging**